### PR TITLE
Remove a few usages of `newDerivedContext`

### PR DESCRIPTION
**Pull request by crazytonyli**
The new "block" parts of the code now run in a background thread, rather than the main thread, which I don't think would have any negative impact for these cases in particular.

The changes are in viewing Reader posts and comments. I didn't find any issue on these areas with these changes.

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
None.

3. What automated tests I added (or what prevented me from doing so)
None.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by mokagio — APPROVED** [review]
Looking good.

After #19427 auto-merges, we should merge `trunk` into this to address the `RELEASE-NOTES.txt` conflict GitHub is reporting.

@crazytonyli if that's okay with you, I'll take care of this. I'm monitoring these PRs closely to start the 21.0 code freeze as soon as CI gives us the okay 😄

**Comment by mokagio** [comment]
> @crazytonyli if that's okay with you, I'll take care of this. I'm monitoring these PRs closely to start the 21.0 code freeze as soon as CI gives us the okay 😄 

I realized I wrote "if that's okay with you," but didn't wait for an answer 😳 . What I should have written was:

> I'll take care of this ... I hope it's okay with you. Let me know if it turns out I made the wrong call and we'll discuss how to do better next time.

**Comment by crazytonyli** [comment]
@mokagio That's okay. Thanks for helping out!

**Pull request merged**
